components/chainhook-sdk/src/chainhooks/bitcoin/mod.rs

```diff
@@ -44,6 +201,177 @@ pub struct BitcoinChainhookPayload {
+    StxTransferred,
+    StxLocked,
+}
+
+#[derive(Clone, Debug, Serialize, Deserialize, Hash, PartialEq, Eq, JsonSchema)]
+#[serde(rename_all = "kebab-case")]
+pub enum OrdinalsMetaProtocol {
+    All,
+    #[serde(rename = "brc-20")]
+    Brc20,
+}
+
+#[derive(Clone, Debug, Serialize, Deserialize, PartialEq, Eq, JsonSchema)]
+pub struct InscriptionFeedData {
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub meta_protocols: Option<HashSet<OrdinalsMetaProtocol>>,
+}
+
+#[derive(Clone, Debug, Serialize, Deserialize, PartialEq, Eq, JsonSchema)]
+#[serde(rename_all = "snake_case", tag = "operation")]
+pub enum OrdinalOperations {
+    InscriptionFeed(InscriptionFeedData),
+}
+
+pub fn get_stacks_canonical_magic_bytes(network: &BitcoinNetwork) -> [u8; 2] {
+    match network {
+        BitcoinNetwork::Mainnet => *b"X2",
+        BitcoinNetwork::Testnet => *b"T2",
+        BitcoinNetwork::Regtest => *b"id",
+        BitcoinNetwork::Signet => unreachable!(),
+    }
+}
+
+pub struct PoxConfig {
+    pub genesis_block_height: u64,
+    pub prepare_phase_len: u64,
+    pub reward_phase_len: u64,
+    pub rewarded_addresses_per_block: usize,
+}
+
+impl PoxConfig {
+    pub fn get_pox_cycle_len(&self) -> u64 {
+        self.prepare_phase_len + self.reward_phase_len
+    }
+
+    pub fn get_pox_cycle_id(&self, block_height: u64) -> u64 {
+        (block_height.saturating_sub(self.genesis_block_height)) / self.get_pox_cycle_len()
+    }
+
+    pub fn get_pos_in_pox_cycle(&self, block_height: u64) -> u64 {
+        (block_height.saturating_sub(self.genesis_block_height)) % self.get_pox_cycle_len()
+    }
+
+    pub fn get_burn_address(&self) -> &str {
+        match self.genesis_block_height {
+            666050 => "1111111111111111111114oLvT2",
+            2000000 => "burn-address-regtest",
+            _ => "burn-address",
+        }
+    }
+}
+const POX_CONFIG_MAINNET: PoxConfig = PoxConfig {
+    genesis_block_height: 666050,
+    prepare_phase_len: 100,
+    reward_phase_len: 2100,
+    rewarded_addresses_per_block: 2,
+};
+
+const POX_CONFIG_TESTNET: PoxConfig = PoxConfig {
+    genesis_block_height: 2000000,
+    prepare_phase_len: 50,
+    reward_phase_len: 1050,
+    rewarded_addresses_per_block: 2,
+};
+
+const POX_CONFIG_DEVNET: PoxConfig = PoxConfig {
+    genesis_block_height: 100,
+    prepare_phase_len: 4,
+    reward_phase_len: 10,
+    rewarded_addresses_per_block: 2,
+};
+
+pub fn get_canonical_pox_config(network: &BitcoinNetwork) -> PoxConfig {
+    match network {
+        BitcoinNetwork::Mainnet => POX_CONFIG_MAINNET,
+        BitcoinNetwork::Testnet => POX_CONFIG_TESTNET,
+        BitcoinNetwork::Regtest => POX_CONFIG_DEVNET,
+        BitcoinNetwork::Signet => unreachable!(),
+    }
+}
+
+#[derive(Debug, Clone, PartialEq)]
+#[repr(u8)]
+pub enum StacksOpcodes {
+    BlockCommit = '[' as u8,
+    KeyRegister = '^' as u8,
+    StackStx = 'x' as u8,
+    PreStx = 'p' as u8,
+    TransferStx = '$' as u8,
```

Comment:
Is there any reason in the codebase to truncate 4 bytes (chars) to 1 byte (u8)? Consider using a byte literal `b'['`, for instance, for those ASCII characters.

Reply:
@MicaiahReid adding clarification to my statements above.

Using a byte literal clarifies the code's intent: only ASCII characters are allowed for `StacksOpcodes`. Any non-ASCII character, like `∞`, will be rejected. Casting types allows to truncate NON-ASCII chars like `'∞' as u8`.

Other thing I am considering in this case is the runtime performance. `StacksOpcodes` is used by the indexer, inside a **loop**. The overhead of casting `'[' as u8` involves a runtime operation to convert the char to u8. This is not a super-heavy operation but is being performed inside a loop in a performance-critical section, thus the difference might become more noticeable.

I'm unsure of the exact impact on the indexer's performance, but any code involving it should consider performance optimization, in my opinion.

Reply:
@csgui Thanks for the explanation! I've applied your suggestion.